### PR TITLE
devcontainer

### PR DESCRIPTION
**Pull request by abernier**
**Description**

This PR aims at enhancing the developer experience, by allowing anyone to have the project up-and-running in a Codespace in one click, right from his browser: https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/setting-up-your-repository/facilitating-quick-creation-and-resumption-of-codespaces

image

This PR adds a `.devcontainer` to the repository, that:

- runs a linux machine (with enough memory)
- with node 18
- automatically being pre-`npm ci`
- launches `npm start`
- opens a http://localhost:8080 preview
- pre-installs eslint extension

NB: It also works locally with [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)

---

- [x] Ready for review
- [ ] let the reviewer runs this devcontainer based on this `abernier:devcontainer` branch: [Open in GitHub Codespaces](https://github.com/codespaces/new?template_repository=abernier%2Fthree.js&ref=devcontainer)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
     "test": "npm run lint && npm run test-unit",
     "build": "rollup -c utils/build/rollup.config.js",
     "build-module": "rollup -c utils/build/rollup.config.js --configOnlyModule",
-    "dev": "concurrently --names \"ROLLUP,HTTP\" -c \"bgBlue.bold,bgGreen.bold\" \"rollup -c utils/build/rollup.config.js -w -m inline\" \"servez -p 8080 --ssl\"",
+    "dev": "concurrently -P --names \"ROLLUP,HTTP\" -c \"bgBlue.bold,bgGreen.bold\" \"rollup -c utils/build/rollup.config.js -w -m inline\" \"servez -p 8080 --ssl {@}\"",
```
Comment (abernier) [inline package.json:46]: this is to allow overriding `servez` options, for `--ssl=false` in https://github.com/mrdoob/three.js/pull/26055/files#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686R9

Comment (LeviPesin) [reply package.json:46]: Why do we need `--ssl=false`?

Comment (abernier) [reply package.json:46]: `--ssl=false` is only for inside the codespace, on localhost, ssl will be on like before

Comment (abernier) [reply package.json:46]: the reason is that the endpoint Codespace mounts for the preview is already configured configured with ssl

**Comment by mrdoob** [comment]
Can the `.devcontainer/devcontainer.json` file go inside `.github/` and avoid polluting the root of the repo?

**Comment by abernier** [comment]
> Can the `.devcontainer/devcontainer.json` file go inside `.github/` and avoid polluting the root of the repo?

Unfortunately I think it cannot: https://containers.dev/implementors/spec/#devcontainerjson

Also because devcontainers are not directly related to github: you can for example execute it on your local vscode with this extension https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers

**Comment by abernier** [comment]
you can have `devcontainer.json` file at root of the project however

**Comment by mrdoob** [comment]
> This PR aims at enhancing the developer experience, by allowing anyone to have the project up-and-running in a Codespace in one click, right from his browser

Do you mind doing a screen recording showing the developer experience?

**Comment by abernier** [comment]
> Do you mind doing a screen recording showing the developer experience?

You can actually try it with that link: https://github.com/codespaces/new?template_repository=abernier%2Fthree.js&ref=devcontainer

**Comment by Mugen87** [comment]
TBH, downloading the repo and executing `npm i && npm start` isn't that complicated. To me, this does not justify yet another service that we are going to use (and maintain). Something like Codespace should work without requiring the repositories to add additional configuration.

**Comment by mrdoob** [comment]
Agreed. The PR didn't have many upvotes either...

**Pull request closed without merging**